packages/flutter_tools/lib/src/device.dart

```diff
@@ -278,6 +279,9 @@ abstract class DeviceManager {
   Future<Device> _chooseOneOfAvailableDevices(List<Device> devices) async {
     _displayDeviceOptions(devices);
     final String userInput =  await _readUserInput(devices.length);
+    if (userInput.toLowerCase() == 'q') {
+      throwToolExit('exit from choose one of available devices', exitCode: 0);
```

Comment:
leave the message blank. exitCode 0 is the default so it isn't necessary either.

Reply:
Are you sure? `throwToolExit('')` exitCode will be `null`

Reply:
and message may help

Reply:
we don't need to tell the user why we exited if they asked to exit

Reply:
```
expect(e.exitCode, 0);
expect(e.message, contains('exit from choose'));
```

with clear test